src/neo/Network/P2P/Payloads/OraclePayload.cs

```diff
@@ -0,0 +1,116 @@
+using Neo.IO;
+using System.IO;
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.Persistence;
+using System;
+using System.Linq;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class OraclePayload : IVerifiable
+    {
+        public byte[] Data;
+        //public uint OracleValidatorIndex;
+        public ECPoint OraclePub;
```

Comment:
For me, I perfer to use pubilc key, as oracle node can be changed during oracle response processing.

Reply:
Agree

Reply:
I also prefer it. On the other side, using `OracleValidatorIndex` makes the messages smaller.